src/main/java/org/candlepin/resource/OwnerProductResource.java

```diff
@@ -150,6 +150,18 @@ public ProductDTO addContent(String ownerKey, String productId, String contentId
     public ProductDTO createProductByOwner(String ownerKey, ProductDTO dto) {
         this.validator.validateCollectionElementsNotNull(
             dto::getBranding, dto::getDependentProductIds, dto::getProductContent);
+        if (dto.getId() == null) {
```

Comment:
Few things here:
- Let's move the simple property checks to before the collection validation, also add whitespace between the checks so they don't look "connected" at a glance (current lack of whitespace matches our normal if-else syntax)
- We can combine the null and empty checks, as it'll reduce the number of strings to translate, and shrinks this block a bit.
- On the note of translations, let's try to use a string that's already present if possible, as it'll avoid making the translators have to translate the same message twice, or build up weird deviations if the verbiage we use doesn't lend itself well to the destination language.
- Don't punctuate the end of the error message, as it's not a complete sentence, and it looks funny when encapsulated. As far as the beginning capitalization goes... we're not at all consistent on that. Even looking at just my own error messages I've been inconsistent with that, so eh... follow your heart. 

Something like this should suffice:
```
if (dto.getId() == null || dto.getId().matches("^\\s*$")) {
  throw new BadRequestException(this.i18n.tr("product has a null or invalid ID"));
}

if (dto.getName() == null || ...
```

Reply:
I checked the /po/keys.pot file for existing keys and it didn't look like there were any existing ones that we can use (I believe that's the correct file to check).  I like the message you provided for the invalid id and I did the same for the name. Those will be very reusable for other locations where product id and name are verified!